test/e2e/performanceprofile/functests/1_performance/performance.go

```diff
@@ -291,22 +290,31 @@ var _ = Describe("[rfe_id:27368][performance]", func() {
 	})
 
 	Context("RPS configuration", func() {
-		It("Should have the correct RPS configuration", func() {
-			if profile.Spec.CPU == nil || profile.Spec.CPU.Reserved != nil {
-				return
+		It("[test_id:55012] Should have the correct RPS configuration", func() {
+			if profile.Spec.CPU == nil || profile.Spec.CPU.Reserved == nil {
+				Skip("Test Skipped due nil Reserved cpus")
 			}
 			if profile.Spec.WorkloadHints != nil && profile.Spec.WorkloadHints.RealTime != nil &&
 				!*profile.Spec.WorkloadHints.RealTime && !profileutil.IsRpsEnabled(profile) {
-				return
+				Skip("realTime Workload Hints is not enabled")
 			}
 
 			expectedRPSCPUs, err := cpuset.Parse(string(*profile.Spec.CPU.Reserved))
 			Expect(err).ToNot(HaveOccurred())
 			for _, node := range workerRTNodes {
 				// Verify the systemd RPS service uses the correct RPS mask
-				cmd := []string{"sed", "-n", "s/^ExecStart=.*echo \\([A-Fa-f0-9]*\\) .*/\\1/p", "/rootfs/etc/systemd/system/update-rps@.service"}
-				serviceRPSCPUs, err := nodes.ExecCommandOnNode(cmd, &node)
+				var maskContent string
+				cmd := []string{"cat", "/rootfs/etc/systemd/system/update-rps@.service"}
+				unitFileContents, err := nodes.ExecCommandOnNode(cmd, &node)
 				Expect(err).ToNot(HaveOccurred())
+				for _, line := range strings.Split(unitFileContents, "\n") {
+					if strings.Contains(line, "ExecStart=/usr/local/bin/set-rps-mask.sh") {
+						maskContent = line
+					}
+				}
+				rpsMaskContent := strings.TrimSuffix(maskContent, "\r")
+				Expect(len(strings.Split(rpsMaskContent, " "))).To(Equal(3), "systemd unit file doesn't have proper rpsmask")
```

Comment:
@mrniranjan what is the expected mask content here ?  will its length always be the constant 3 ?

Reply:
@yanirq Yes, Example ["ExecStart=/usr/local/bin/set-rps-mask.sh", "%i", "0000001f"]

Reply:
maybe change the error message to :
systemd unit file rps mask value is empty

Reply:
ack